codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -790,66 +775,83 @@ let invalidatedActionReducer = (state: t, action: action) =>
     (state, [])
   }
 
-let executePartitionQuery = (
-  query,
+let executeQuery = (
+  query: FetchState.query,
   ~logger,
   ~chainWorker,
   ~currentBlockHeight,
   ~chain,
-  ~dispatchAction,
   ~isPreRegisteringDynamicContracts,
 ) => {
-  let module(ChainWorker: ChainWorker.S) = chainWorker
-
-  let logger = Logging.createChildFrom(
-    ~logger,
-    ~params={
-      "chainId": chain->ChainMap.Chain.toChainId,
-      "logType": "Block Range Query",
-      "workerType": ChainWorker.name,
-    },
-  )
-  let logger = query->FetchState.getQueryLogger(~logger)
-  ChainWorker.fetchBlockRange(
-    ~query,
-    ~logger,
-    ~currentBlockHeight,
-    ~isPreRegisteringDynamicContracts,
-  )->Promise.thenResolve(res =>
-    switch res {
-    | Ok(res) => dispatchAction(BlockRangeResponse(chain, res))
-    | Error(e) => dispatchAction(ErrorExit(e))
-    }
-  )
+  switch query {
+  | PartitionQuery({partitionId, fromBlock, toBlock, contractAddressMapping}) =>
+    chainWorker->ChainWorker.fetchBlockRange(
+      ~fromBlock,
+      ~toBlock,
+      ~contractAddressMapping,
+      ~partitionId,
+      ~chain,
+      ~currentBlockHeight,
+      ~isPreRegisteringDynamicContracts,
+      ~logger,
+      //Only apply wildcards on the first partition
+      //to avoid duplicate wildcard queries
+      ~shouldApplyWildcards=partitionId === "0",
+    )
+  | MergeQuery({partitionId, fromBlock, toBlock, contractAddressMapping}) =>
+    chainWorker->ChainWorker.fetchBlockRange(
+      ~fromBlock,
+      ~toBlock=Some(toBlock),
+      ~contractAddressMapping,
+      ~partitionId,
+      ~chain,
+      ~currentBlockHeight,
+      ~isPreRegisteringDynamicContracts,
+      ~logger,
+      //Only apply wildcards on the first partition
+      //to avoid duplicate wildcard queries
+      ~shouldApplyWildcards=partitionId === "0",
```

Comment:
These two seem so similar it looks like we should be able to have more shared data in query. Maybe even just currying the function above would look better 😝